coala_quickstart/green_mode/file_aggregator.py

```diff
@@ -0,0 +1,81 @@
+import os
+
+from coala_quickstart.green_mode.find_globs import (
+    find_globs_from_files,
+    )
+
+
+class Node:
+    """
+    Tree data-structure that has each directory as a node,
+    each subdirectory as children and files as leaves.
+    """
+
+    def __init__(self, name, parent, children=[]):
+        self.name = name
+        self.parent = parent
+        self.children = children
+
+    def add_children(self, children):
+        self.children.append(children)
+
+    def search_name(self, name):
+        if name == self.name:  # pragma: no cover
```

Comment:
The test case for this seems simple, why is this no cover?

Reply:
it sure *"seems"* simple but if you could just have a look at `file_aggregatorTest.py` you will find they are actually quite difficult and can be tested when I have a complex file directory structure inside just for testing. @jayvdb suggested a few methods and I would be looking into them after GSoC.